modin/config/pubsub.py

```diff
@@ -13,9 +13,80 @@
 
 """Module houses ``Parameter`` class - base class for all configs."""
 
+import warnings
 from collections import defaultdict
 from enum import IntEnum
-from typing import Any, Callable, DefaultDict, NamedTuple, Optional, Tuple
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    DefaultDict,
+    NamedTuple,
+    Optional,
+    Tuple,
+    cast,
+)
+
+if TYPE_CHECKING:
+    from .envvars import EnvironmentVariable
+
+
+class DeprecationDescriptor:
+    """
+    Describe deprecated parameter.
+
+    Parameters
+    ----------
+    parameter : type[Parameter]
+        Deprecated parameter.
+    new_parameter : type[Parameter], optional
+        If there's a replacement parameter for the deprecated one, specify it here.
+    when_removed : str, optional
+        If known, the exact release when the deprecated parameter is planned to be removed.
+    """
+
+    _parameter: type["Parameter"]
+    _new_parameter: Optional[type["Parameter"]]
+    _when_removed: str
+
+    def __init__(
+        self,
+        parameter: type["Parameter"],
+        new_parameter: Optional[type["Parameter"]] = None,
+        when_removed: Optional[str] = None,
+    ):
+        self._parameter = parameter
+        self._new_parameter = new_parameter
+        self._when_removed = "a future" if when_removed is None else when_removed
+
+    def deprecation_message(self, use_envvar_names: bool = False) -> str:
+        """
+        Generate a message to be used in a warning raised when using the deprecated parameter.
+
+        Parameters
+        ----------
+        use_envvar_names : bool, default: False
+            Whether to use environment variable names in the warning. If ``True``, both
+            ``self._parameter`` and ``self._new_parameter`` have to be a type of ``EnvironmentVariable``.
+
+        Returns
+        -------
+        str
+        """
+        name = (
```

Comment:
```suggestion
        if use_envvar_names:
            from .envvars import EnvironmentVariable

            if not isinstance(self._parameter, EnvironmentVariable):
                raise ValueError("'use_envvar_names==True' can be used only with 'EnvironmentVariable' type.")
        name = (
```

Reply:
In this case, let's show intentions explicitly.

Reply:
Trying to satisfy flake8, black and mypy's requirements for this piece of code resulted into this mess:
```python
if use_envvar_names:
    from .envvars import (  # noqa : F811 (redefinition of unused variable)
        EnvironmentVariable,
    )

    if not isinstance(
        # 'cast' here, as otherwise mypy complains, that this condition always gives 'False'...........
        cast("EnvironmentVariable", self._parameter), EnvironmentVariable
    ) or (
        self._new_parameter is not None
        and not isinstance(
            cast("EnvironmentVariable", self._new_parameter),
            EnvironmentVariable,
        )
    ):
        raise ValueError(
            "'use_envvar_names==True' can be used only with 'EnvironmentVariable' type."
        )
```
I don't really want to put this overcomplicated stuff into the code, knowing that this safeguard will never ever shoot, as we don't have other vars rather than EnvVars, but only distract from the actual logic of the method

Reply:
ok